src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        T value;
+        Node<T> next;
+        Node<T> prev;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    public MyLinkedList() {
+        this.size = 0;
+    }
+
     @Override
```

Comment:
Private methods and classes should be after public ones in your class. Move the `Node<T>` class definition to the end of the `MyLinkedList` class.